leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/CaliforniumObservationRegistryImpl.java

```diff
@@ -64,6 +66,7 @@
 
     private final List<ObservationRegistryListener> listeners = new CopyOnWriteArrayList<>();
     private final Map<KeyToken, Observation> observationsByToken = new ConcurrentHashMap<>();
+    private final Lock observationLock = new ReentrantLock();
```

Comment:
There is no good reason to use a lock here, you can use a private monitor. "private Object monitor = new Object();" if you really want to isolate it.

A synchronize on monitor is superior here because you don't have to rely on the try/finally and you are not using the extra feature of Lock missing from synchronize block.

That's no religious/style thing, just better idiomatic java :smile: